### PR TITLE
Add __init__ stub for CaseInsensitiveDict

### PR DESCRIPTION
**Pull request by adamtheturtle**
The error I want to avoid:

```python
from requests.structures import CaseInsensitiveDict

case_insensitive_dict = CaseInsensitiveDict({})
case_insensitive_dict_2 = CaseInsensitiveDict(data={})
```

```
example.py:3: error: Too many arguments for "CaseInsensitiveDict"
example.py:4: error: Unexpected keyword argument "data" for "CaseInsensitiveDict"
```

I am not 100% sure on the types that should be supported here, and am not very familiar with `Text` or `MutableMapping` but they seem to be used in similar places in this directory.
I'm happy to change this given feedback.

### CODE REVIEW

File: third_party/2and3/requests/structures.pyi [hunk 1]
```diff
@@ -1,6 +1,8 @@
 from typing import Any, Iterator, MutableMapping, Text, Tuple, Union
 
 class CaseInsensitiveDict(MutableMapping[str, Union[Text, bytes]]):
+    def __init__(self, data=Optional[
+        MutableMapping[Union[str, bytes], Any]]) -> None: ...
```
Comment (gvanrossum) [inline third_party/2and3/requests/structures.pyi]: Why did you make the key type `Union[str, bytes]`? It seems pretty clear that the other methods only accept keys for dicts. Similar for the value type -- why not `Union[Text, bytes]` like the rest of the API here?

Comment (JelleZijlstra) [reply third_party/2and3/requests/structures.pyi]: Also, you're making `Optional[...]` the default, value, not the type. You want something like `data: Optional[...] = ...`.

Why do you need a MutableMapping? I would expect it to accept any Mapping.

**Comment by JelleZijlstra** [comment]
Let's merge #1897 instead, which fixes a few more issues in the same file.

**Comment by adamtheturtle** [comment]
@JelleZijlstra Thanks and sorry I lost track of this!

**Pull request closed without merging**